pkg/fqdn/dnsproxy/proxy.go

```diff
@@ -578,6 +578,8 @@ func StartDNSProxy(
 	notifyFunc NotifyOnDNSMsgFunc,
 	concurrencyLimit int, concurrencyGracePeriod time.Duration,
 ) (*DNSProxy, error) {
+	// This is no longer used by the dnsproxy itself, but other code paths are still using it
+	// TODO GH-22241 for more information and discussion
```

Comment:
On that note, what if people disable the dnsproxy by using `--enable-l7-proxy=false`, will the other usage of the lru cause issues when its not initialized? Started thinking about it now, so haven't had time to test it yet.

Reply:
I think we are OK because the check for `enable-l7-proxy` will prevent the processing of any policies with any L7 rules. This should be pretty easy to verify. Apply a FQDN policy with the `enable-l7-proxy` disabled. You should see a msg saying that the policy was not applied in the logs and the Agent doesn't panic (assuming that's what would happen when trying to insert into a non-init'd LRU).

Reply:
Ahh, makes sense, thanks!

Reply:
Shouldn't we move the InitRegexCompileLRU call out of the dnsproxy package in that case? Since this is initializing a global cache.

Reply:
And presumably tie it to the `DNSProxy` object instead?

Reply:
I don't think we need it at all here (judging from the comment), just move the init up to the daemon package.